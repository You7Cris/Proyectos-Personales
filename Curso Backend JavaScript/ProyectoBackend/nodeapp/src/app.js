// Paso 3
const express = require('express');
const config = require('./config');

const app = express();
app.set('port', config.app.port);

// Paso 20
// Cargo en la app las rutas para los servicios o modulos de la app
// const gestion_juego = require('./modulos/gestion_juegos/rutas');
const gestion_juego = require('./modulos/gestion_juegos/rutas');
const gestion_genero = require('./modulos/gestion_generos/rutas');
const gestion_grupo = require('./modulos/gestion_grupos/rutas');


// Paso 21
// Creo las rutas a los diferentes servicios o modulos de la app
app.use('/api/gestion_juego', gestion_juego);
app.use('/api/gestion_genero', gestion_genero);
app.use('/api/gestion_grupo', gestion_grupo);

// Paso 22
//localhost:4000/api/gestion_juego

// Paso 6
module.exports = app;
