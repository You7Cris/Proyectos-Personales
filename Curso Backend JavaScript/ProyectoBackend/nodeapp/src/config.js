//Paso 25
// Se carga a dotenv para manejar las variables de entorno y asi poder utilizar el valor de "process.env.PORT" del atributo "port"

require('dotenv').config();


module.exports = {

    //Paso 1 
    // Se crea el atributo "port" para definirr el puerto por el que se podra realizar la conexion a la app

    app:{
        port: process.env.PORT || 4000,
    },

    //Paso 30 se configuran los atributos requeridos para la conexion con la BD
    mysql:{
        host: process.env.DB_HOST || 'localhost',
        user: process.env.DB_USER || 'root',
        passwd: process.env.DB_PASS || '',
        database: process.env.DB || 'db_asignacion_juegos',
    }

    //continua en mysql.js
}

//Paso 2
//mediante la consol√±a de comandos se ingresa a la carpeta del proyecto y desde allio se instalan
// express, nodemon y dotenv -D
//Continua en app.js