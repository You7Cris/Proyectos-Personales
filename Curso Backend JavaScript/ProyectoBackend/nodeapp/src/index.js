
//Paso 7
// la instancia de la aplicacion

const app = require('./app');

//Paso 8
// Inicializamos el servidor 
// Con el listen escucha el puerto
// La funcion flecha para revisar que este bien

app.listen(app.get('port'), () => {
    console.log('Server on port', app.get('port'));
});