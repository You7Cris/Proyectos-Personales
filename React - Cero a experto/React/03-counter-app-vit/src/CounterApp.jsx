import { useState } from 'react'; // esto es un Hook
import PropTypes from 'prop-types'

export const CounterApp = ( {value} ) => {

    console.log('render');
    // API....FETCH.......

    const [ counter, setCounter ] = useState( value );

    const hanleAdd = () => { 
        //console.log(event)
        //console.log(newValue)
        //counter++;
        setCounter( counter + 1 );
        //setCounter( (c) => c + 1 )
    }

    // const restAdd = () => {
    //     setCounter( counter - 1 );
    // }

    //const handleSubstract = () => setCounter( counter - 1);
    //const handleReset = () => setCounter( value );
    const restAdd = () => setCounter( counter - 1);
    const resetAdd = () => setCounter( value );

    // const resetAdd = () => {
    //     setCounter( value );
    // }
    

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={ hanleAdd }> +1 </button>
            <button onClick={ restAdd }> -1 </button>
            <button onClick={ resetAdd }> Reset </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}