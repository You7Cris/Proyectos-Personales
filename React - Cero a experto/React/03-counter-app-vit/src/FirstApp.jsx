//import { Fragment } from 'react';

//const newMessage = 'Cristian';
//const newMessage = [1,2,3,4,5,6,7,8,9];
// const newMessage = {
//     message: 'Hola Mundo',
//     title: 'Cristian'
// };

// Async daria error
// const getResult = (a, b) => {
//     return a + b;
// }

import PropTypes from 'prop-types';



// props -> propiedades que le estamos mandando a la funcion
export const FirstApp = ( /*props*/ { title, subtitle, name } ) => {

    //console.log(props);
    // if ( !title ) {
    //     throw new Error('El titulo no existe');
    // }

    return (
        <>
        {/* <div> */}
            {/* <h1>{ newMessage }</h1> */}
            {/* <h1>{ JSON.stringify(newMessage) }</h1> */}
            {/* <h1>{ getResult(5, 9) }</h1> */}
            {/* <h1>{ title }</h1> */}
            <h1 data-testid="test-title">{ title }</h1>
            <p>{ subtitle }</p>
            <h1>{ subtitle }</h1>
            <p>{ name }</p>
            {/* <p>Soy un subtitulo</p> */}
        {/* </div> */}
        </>

    )
    
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    //title: 'No hay titulo',
    subtitle: 'No hay subtitulo' ,
    name: 'Cristian Gonzalez',
}