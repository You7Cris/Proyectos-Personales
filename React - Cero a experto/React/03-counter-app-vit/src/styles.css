html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 10px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
    margin: 15px;
}

footer {
    text-align: center;
    font-size: 0.8rem;
    margin-top: 15px;
}