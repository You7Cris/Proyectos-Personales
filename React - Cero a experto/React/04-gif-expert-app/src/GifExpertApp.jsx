import React, { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState([ 'One Punch' ]);

    // console.log(categories);
    const onAddCategory = (newCategory) => {
        console.log(newCategory);

        if ( categories.includes(newCategory) ) return;
        //categories.push(newCategory);
        // categories.push('Valorant');
        // setCategories( [ ...categories, 'Valorant' ]);
        //setCategories( cat => [...cat, 'Valorant']);
        setCategories( [ newCategory, ...categories]);
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory 
                //setCategories={ setCategories }
                onNewCategory={ (value) => onAddCategory(value)}
                //currentCategories={ categories }
            />
            {/* Listado de items (Gifs) */}
           
            { 
                categories.map( ( category ) => 
                    (
                        <GifGrid 
                            key={ category } 
                            category={ category } />
                )) 
            }
            {/* <li>ABC</li> */}
            
            {/* Gif Item */}
        </>
    )
}