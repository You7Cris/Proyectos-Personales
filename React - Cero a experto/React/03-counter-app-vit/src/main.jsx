import React from 'react' //imr importa directamente react
import ReactDOM  from 'react-dom/client'
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
//import { App } from './HelloWorldApp'
//import HelloWorldApp from './HelloWorldApp'

import './styles.css';
import { CounterApp } from './CounterApp';




ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        {/* <App /> */}
        {/* <HelloWorldApp /> */}
        {/* <FirstApp title = "Hola, Soy Cristian" subtitle= { 123 }/> */}
        {/* <FirstApp title="Hola, Soy Cristian" />  */}
        < CounterApp value={1} />
    </React.StrictMode>
);