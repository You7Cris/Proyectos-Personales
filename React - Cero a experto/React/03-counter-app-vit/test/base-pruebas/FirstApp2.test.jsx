import { render, screen } from '@testing-library/react';
import { FirstApp } from '../../src/FirstApp';

describe('Pruebas en < FirstApp />', () => {

  const title = "Hola, Soy Goku";
  const subTitle = "Soy un subtitulo";

  test('debe de hacer math con el snapshot', () => {
    
    const { container } = render( <FirstApp title={ title } />);
    expect( container ).toMatchSnapshot();
  });

  test('debe de mostrar el mensaje "Hola, Soy Goku" ', () => {

    
    render( <FirstApp title={ title } />);
    // screen.debug();
    expect( screen.getByText(title) ).toBeTruthy();

  });

  test('debe de mostrar el titulo en un h1', () => {
    render( <FirstApp title={ title } />);
    expect( screen.getAllByRole('heading', { level: 1}).innerHTML ).toContain( title );
  });

  test('debe de mostrar el subtitulo enviado por props', () => {

    render( <FirstApp title={ title } subtitle={ subTitle } />);

    expect( screen.getAllByText(subtitle).length).toBe(2);
    
  })
  
  
  
  
  
  
  
});
