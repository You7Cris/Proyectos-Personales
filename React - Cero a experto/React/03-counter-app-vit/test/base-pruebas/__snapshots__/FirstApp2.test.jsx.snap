// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en < FirstApp /> debe de hacer math con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola, Soy Goku
  </div>
  <p>
    No hay subtitulo
  </p>
  <h1>
    No hay subtitulo
  </h1>
  <p>
    Cristian Gonzalez
  </p>
</div>
`;
